app/jobs/cleanup_working_file_job.rb

```diff
@@ -13,13 +13,14 @@
 # ---  END LICENSE_HEADER BLOCK  ---
 
 class CleanupWorkingFileJob < ActiveJob::Base
-  def perform(masterfile_id)
-    masterfile = MasterFile.find(masterfile_id)
-    masterfile.working_file_path.map do |path|
+  def perform(masterfile_id, working_file_path)
+    working_file_path.map do |path|
       parent_directory = File.dirname(path)
       File.delete(path) if File.exist?(path)
       Dir.delete(parent_directory) if Dir.exist?(parent_directory)
     end
+
+    masterfile = MasterFile.find(masterfile_id)
     masterfile.working_file_path = nil
```

Comment:
Do we care if the job still fails here if the mf is gone?

Reply:
I think it should still raise some error in case the master file wasn't intentionally deleted.